src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -99,18 +99,51 @@
 
     @Override
     public Connection connect(String url, Properties info) throws SQLException {
+        Pair<MongoConnection, Integer> p = getUnvalidatedConnectionAndTimeout(url, info);
+        Connection conn = p.left();
+        conn.isValid(p.right());
+        return conn;
+    }
+
+    public MongoConnection createUnvalidatedConnection(String url, Properties info)
+        throws SQLException {
+        return getUnvalidatedConnectionAndTimeout(url, info).left();
+    }
+
+    private Pair<MongoConnection, Integer> getUnvalidatedConnectionAndTimeout(String url, Properties info)
+        throws SQLException {
         if (!acceptsURL(url)) {
             return null;
         }
         if (info == null) {
             info = new Properties();
         }
+
         // reuse the code getPropertyInfo to make sure the URI is properly set wrt the passed
         // Properties info value.
         Pair<ConnectionString, DriverPropertyInfo[]> p = getConnectionString(url, info);
+        // ensure that the ConnectionString and Properties are consistent.
+        reconcileProperties(p.right(), info);
+
+        ConnectionString cs = p.left();
+        Connection ret = createDialectConnection(cs, info);
+        System.out.println(cs);
```

Comment:
This may have been left in from debugging.  The cs at this point may have the user/pass so we don't want to print it.

Reply:
Definitely not!